volmdlr/wires.py

```diff
@@ -634,6 +635,72 @@ def random_point_inside(self):
     #     else:
     #         raise NotImplementedError('Non convex contour not supported yet')
 
+    @classmethod
+    def add_points_on_contour(cls, contour, points: List[volmdlr.Point3D]):
+        for primitive in contour.primitives:
+            if not isinstance(primitive, volmdlr.edges.LineSegment2D):
+                raise KeyError('primitives must be define with only LineSegment2D')
+
+        primitives = [p for p in contour.primitives]
+        for point in points:
+            dist_min = math.inf
+            for primitive in primitives:
+                dist = primitive.point_distance(point)
+                if dist < dist_min:
+                    dist_min = dist
+                    prim_opt = primitive
+            if dist_min > 1e-15:
+                continue
+            new_primitives = []
+            for primitive in primitives:
+                if prim_opt.start != point and prim_opt.end != point and primitive == prim_opt:
+                    new_primitives.append(volmdlr.edges.LineSegment2D(prim_opt.start, point))
+                    new_primitives.append(volmdlr.edges.LineSegment2D(prim_opt.end, point))
+                elif primitive != prim_opt:
+                    new_primitives.append(primitive)
+            primitives = new_primitives
+        return cls(primitives)
```

Comment:
There are methods to transform to polygon like polygon_points

Reply:
did you look at polygon points? I dont like this function because it is particular to polygons